pkg/codegen/pcl/binder_schema.go

```diff
@@ -566,8 +567,44 @@ func GenEnum(
 		contract.Assertf(ok,
 			"We have determined %s is a safe enum, which we define as "+
 				"being able to calculate a member for", t)
-		safeEnum(member)
+		if member != nil {
+			safeEnum(member)
+		} else {
+			unsafeEnum(from)
+			knownVal := strings.Split(strings.Split(known.GoString(), "(")[1], ")")[0]
+			return &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  fmt.Sprintf("%v is not a valid value of the enum \"%v\"", knownVal, t.Token),
+				Detail:   fmt.Sprintf("Valid members are %v", listToString(enumMemberValues(t))),
+			}
```

Comment:
`enumMemberValues(t)` could be `nil` or empty. We don't want to print `"Valid members are "` 😃. 

```suggestion
			diag := &hcl.Diagnostic{
				Severity: hcl.DiagError,
				Summary:  fmt.Sprintf("%v is not a valid value of the enum \"%v\"", knownVal, t.Token),
			}
			if members := enumMemberValues(t); len(members) > 0 {
				diag.Detail = fmt.Sprintf("Valid members are %v", listToString(enumMemberValues(t))),
			}
			return diag
```